src/compiler/program.ts

```diff
@@ -587,6 +594,7 @@ namespace ts {
         let moduleResolutionCache: ModuleResolutionCache;
         let resolveModuleNamesWorker: (moduleNames: string[], containingFile: string) => ResolvedModuleFull[];
         const hasInvalidatedResolution = host.hasInvalidatedResolution || returnFalse;
+        const hasChangedAutomaticTypeDirectiveNames = host.hasChangedAutomaticTypeDirectiveNames && host.hasChangedAutomaticTypeDirectiveNames.bind(host) || returnFalse;

```

Comment:
Do we expect this value to change *during* the call to `createProgram`? Otherwise you could just call that function once and store it as a boolean rather than as a function.